js/compiler/compiler.go

```diff
@@ -44,103 +44,119 @@ var (
 		"highlightCode": false,
 	}
 
-	compilerInstance *Compiler
-	once             sync.Once
+	once sync.Once // nolint:gochecknoglobals
+	babl *babel    // nolint:gochecknoglobals
 )
 
-// A Compiler uses Babel to compile ES6 code into something ES5-compatible.
-type Compiler struct {
-	vm *goja.Runtime
+// CompatibilityMode specifies the JS compatibility mode
+// nolint:lll
+//go:generate enumer -type=CompatibilityMode -transform=snake -trimprefix CompatibilityMode -output compatibility_mode_gen.go
+type CompatibilityMode uint8
 
-	// JS pointers.
-	this      goja.Value
-	transform goja.Callable
-	mutex     sync.Mutex //TODO: cache goja.CompileAST() in an init() function?
-}
-
-// Constructs a new compiler.
-func New() (*Compiler, error) {
-	var err error
-	once.Do(func() {
-		compilerInstance, err = new()
-	})
-
-	return compilerInstance, err
-}
-
-func new() (*Compiler, error) {
-	conf := rice.Config{
-		LocateOrder: []rice.LocateMethod{rice.LocateEmbedded},
-	}
-
-	babelSrc := conf.MustFindBox("lib").MustString("babel.min.js")
-
-	c := &Compiler{vm: goja.New()}
-	if _, err := c.vm.RunString(babelSrc); err != nil {
-		return nil, err
-	}
+const (
+	// CompatibilityModeES6 is achieved with Babel and core.js
+	CompatibilityModeES6 CompatibilityMode = iota + 1
+	// CompatibilityModeES51 is standard goja
+	CompatibilityModeES51
+)
 
-	c.this = c.vm.Get("Babel")
-	thisObj := c.this.ToObject(c.vm)
-	if err := c.vm.ExportTo(thisObj.Get("transform"), &c.transform); err != nil {
-		return nil, err
-	}
+// A Compiler compiles JavaScript source code (ES5.1 or ES6) into a goja.Program
+type Compiler struct{}
 
-	return c, nil
+// New returns a new Compiler
+func New() *Compiler {
+	return &Compiler{}
 }
```

Comment:
I decided to leave the initializer to avoid breaking the API too much, and we might use it in the future.